Scripts/Runtime/Entities/Util/WorldUtil.cs

```diff
@@ -0,0 +1,230 @@
+
+    /// <summary>
+    /// A system group to house a <see cref="World"/>s <see cref="EndSimulationEntityCommandBufferSystem"/>.
+    /// Used for mutli-world playerloop optimization.
+    /// </summary>
+    /// <remarks>Kept outside <see cref="WorldUtil"/> so their names read better in the editor.</remarks>
+    [DisableAutoCreation]
+    internal class EndSimulationCommandBufferSystemGroup : ComponentSystemGroup { }
+
+    /// <summary>
+    /// A <see cref="PlayerLoop"/> phase inserted immediately after <see cref="Update"/>.
+    /// Add via <see cref="WorldUtil.AddCustomPhasesToCurrentPlayerLoop" />.
+    /// </summary>
+    internal class PostUpdate
+    {
+        /// <summary>
+        /// Provides a <see cref="PlayerLoopSystem" /> instance to to identify this phase.
+        /// </summary>
+        public static PlayerLoopSystem PlayerLoopSystem
+        {
+            get => new PlayerLoopSystem()
+            {
+                type = typeof(PostUpdate)
+            };
+        }
+    }
+
+    /// <summary>
+    /// A <see cref="PlayerLoop"/> phase inserted immediately after <see cref="Initialization"/>.
+    /// Add via <see cref="WorldUtil.AddCustomPhasesToCurrentPlayerLoop" />.
+    /// </summary>
+    internal class PostInitialization
+    {
+        /// <summary>
+        /// Provides a <see cref="PlayerLoopSystem" /> instance to to identify this phase.
+        /// </summary>
+        public static PlayerLoopSystem PlayerLoopSystem
+        {
+            get => new PlayerLoopSystem()
+            {
+                type = typeof(PostInitialization)
+            };
+        }
+    }
+
+    /// <summary>
+    /// A collection of utilities to manipulate and augments <see cref="World"/>s.
+    /// The Anvil compliment to <see cref="ScriptBehaviourUpdateOrder"/>.
+    /// </summary>
+    public static class WorldUtil
+    {
+        // Not need to reset between play sessions because PlayerLoop systems are stateless and 
+        // persist between sessions when domain reloading is disabled.
+        private static bool s_AreCustomPlayerLoopPhasesAdded = false;
+
+        /// <summary>
+        /// Add custom phases to the <see cref="PlayerLoop"/>.
+        /// <see cref="PostInitialization"/> and <see cref="PostUpdate"/>.
+        /// </summary>
+        public static void AddCustomPhasesToCurrentPlayerLoop()
+        {
+            if (s_AreCustomPlayerLoopPhasesAdded)
+            {
+                return;
+            }
+
+            // Create new top level player loop phases
+            PlayerLoopSystem playerLoop = PlayerLoop.GetCurrentPlayerLoop();
+            List<PlayerLoopSystem> topLevelPhases = playerLoop.subSystemList.ToList();
+
+            int initializationPhaseIndex = topLevelPhases.FindIndex((phase) => phase.type == typeof(Initialization));
+            Debug.Assert(initializationPhaseIndex != -1, "Initialization phase not found");
```

Comment:
```suggestion
            Debug.Assert(initializationPhaseIndex != -1, $"{typeof(Initialization)} phase not found");
```

Reply:
edited to `nameof` instead of `typeof`...you still get the credit :)